### PR TITLE
Don't drop property for `malli.util/update`

### LINKED ISSUE #412: malli.util/update drops the properties of the key it updates
When using `malli.util/update` to update a key in a map schema like so: 

```
(malli.util/update
   (malli.core/schema
      [:map
       [:a {:optional true}
        int?]
       [:b {:optional true} int?]])
    :a identity)
```

The key that gets updated loses its properties. 

`;=> [:map [:a int?] [:b {:optional true} int?]]`

This is because `malli.util/update` is implemented as a get and set, it doesn't retrieve the properties.

### PR DESCRIPTION
**Pull request by ilmoraunio**
Fixes #412, a breaking change.

### CODE REVIEW

**Review by ikitommi — COMMENTED** [review]
how does this work with functions using path vectors like `(mu/update-in ... [:a :b :c] ...)`. Test to document this would be welcome

File: src/malli/core.cljc [hunk 1]
```diff
@@ -177,7 +177,16 @@
 (defn -set-entries [schema key value]
   (let [found (atom nil)
         [key :as new-child] (if (vector? key) (conj key value) [key value])
-        children (cond-> (mapv (fn [[k :as child]] (if (= key k) (do (reset! found true) new-child) child)) (-children schema))
+        new-props (when (= (count new-child) 3) (nth new-child 1))
+        children (cond-> (mapv (fn [[k props :as child]]
```
Comment (ikitommi) [inline src/malli/core.cljc:181]: what about format `[:user/id {:optional true}]|`? are the entries passed in as tuple-3 here? If, no need to check the length, if not, this doesn't work correctly

Comment (ilmoraunio) [reply src/malli/core.cljc:181]: Simplified the check a bit

File: src/malli/core.cljc [hunk 2]
```diff
@@ -177,7 +177,16 @@
...
+                                 (if (= key k)
+                                   (do (reset! found true)
+                                       (cond-> [key]
+                                         new-props (conj new-props)
+                                         (and (not new-props) props) (conj props)
```
[the remark this answers is not in the document]
Comment (ilmoraunio) [reply src/malli/core.cljc:186]: not sure what you mean, `cond->` doesn't really short circuit its test expressions and we still need to check that we aren't putting in both props.

**Comment by ilmoraunio** [comment]
Documented `mu/update-in` behavior via tests, seems to work consistently?

File: src/malli/core.cljc (hunk 2 above)
[the remark this answers is not in the document]
Comment (ikitommi) [reply src/malli/core.cljc:186]: oh, was not `cond`, my bad.

**Review by ikitommi — APPROVED** [review]
Looks good. As we break/fix the current behaviour, we could also cover a case of "reset properties": given an vector of just key in path, there is no `new-properties` and we should remove the props completely:

```clj
(mu/update-in
  [:map
   [:a {:optional true} [:map
                         [:a {:optional true} :int]]]]
  [[:a] :a]
  (constantly :string))
;[:map
; [:a [:map
;      [:a {:optional true} :int]]]]
```

As it's not part of the original issue, can be done separately, what do you think?

**Comment by ikitommi** [comment]
... actually, the "reset properties" works now, so should be fixed now:

```clj
(mu/update-in
  [:map
   [:a {:optional true} [:map
                         [:a {:optional true} :int]]]]
  [[:a] :a]
  (constantly :string))
; => [:map [:a [:map [:a :string]]]]
```

**Review by ikitommi — CHANGES_REQUESTED** [review]
the "reset properties"

**Comment by ikitommi** [comment]
@ilmoraunio would like to merge this, do you have time to fix the `reset properties`? can do that too.

**Comment by ikitommi** [comment]
will fix the last issue, merging this, thanks @ilmoraunio !

**Pull request merged**